### PR TITLE
Add two-piece and telescoping fishing rods

### LINKED ISSUE #56474: Addition of telescopic fishing rods
### Is your feature request related to a problem? Please describe.

Telescoping fishing rods, which are much more compact compared to 2 meter long pro fishing rods currently do not exist in the game. With a double action for telescoping fishing rods, one being to fish, other to fold/unfold, could make wilderness life more viable. They might have slightly lower efficiency compared to pro ones, but their compact nature makes them much more efficient.

### Solution you would like.

Adding telescoping fishing rods to game.

### Describe alternatives you have considered.

Hauling a 2 meter pro fishing rod on a spear strap.

### Additional context

Into the wild people rejoice!

### PR DESCRIPTION
**Pull request by cake-pie**
#### Summary
Content "Add two-piece and telescoping fishing rods"

#### Purpose of change
Resolves #56474 
:warning: Requires #62945

The pro fishing rod has the best fishing tool quality, but is also incredibly unwieldy, inconvenient to transport and store.
More portable options exist in the real world and should be available in-game.

Also updates melee `to_hit` stat for the existing fishing rods to newer format

##### Core changes
- add two-piece and telescoping fishing rods, with same tool quality as the pro fishing rod
- add disassembly for new fishing rods
- permit NPCs to use full range of fishing rods instead of only two specific items
- spawn new fishing rods in relevant locations (see detail below)

##### Item spawning
- created item group `tools_fishingrods_commercial` representing what you'd expect when referring to "commercially available fishing rod" in general.  Apply this in pawn shop, junk trader, college lake, home hardware, general sports items
- created item group `tools_fishingrods_portable` for applications where portability and the ability to pack into a more space-saving form factor is important, like in itemgroups of things that may be brought along on a hiking or camping trip.
  - the hobo reel seems like more of a niche thing IRL but I'd feel bad removing it almost entirely, so giving it a (somewhat disproportionate) probability to show up in the portable group.
- church NPC Kostas personal belongings is inteded to reflect "camping itemgroup modified" with removal of junk, common items, and things like professional fishing rod which were too bulky/long. It will now have a chance to draw one of the portable rods.
- item group `fishing_items` changed to no longer includes makeshift items like bone fishing hook, basic fishing rod (a crude "stick with a piece of string and a hook."). These would not have been common prior to the cataclysm and it would only make sense for them to appear in the context of survivor NPCs (personal belongings, or in a place they've inhabited for some time). Definitely shouldn't be appearing among abandoned pre-cataclysm stuff in places like farm supply store, among the belongings of lake research station staff, or random outdoor "gone fishing" mapgen.
  - item group `tools_fishing_makeshift` created in the hope it may be useful for specific situations where they show up as NPC's belongings, or in places where other survivors may have left them.

#### Misc auxiliary changes
- on existing `fishing_rod_basic` and `fishing_rod_professional`, change `to_hit` stat from legacy to newer format.
- in basecamp fishing recipes groups, adjust description of fish trapping to reflect that both plastic fish trap and makeshift basket trap can be used (746d0f7c4b8d6464924d8c7334e5761f3812d33e)
- sort out ordering of fishing/software NPC recipes which were interleaved with one another
- misc spell check

#### Describe alternatives you've considered
More detailed implementations such as:
- Instead of using same tier as pro fishing rod, fine tune tool quality of rods to account for details about the fishing rod characteristics like length, "action" (jargon: how much they flex), sensitivity, etc. that affect fishing success rate due to impact on factors like how far you can accurately cast and how well you can detect when a fish bites.
- Complex system similar to gunmods so that on top of a base fishing rod of whichever type we can have interchangeable parts: reel, line, hook, sinker, bait/lure, etc. These may affect what fish might be caught, e.g. appropriately sized hook; suitable bait/lure; line rating caps max weight (and may snap if overloaded)....

Would require far more work and too complex for our purposes; this isn't a fishing simulator.

#### Testing
Debug spawn the new fishing rods, test ability to assemble/disassemble, extend/retract and fish with them; place on ground to check that they are using correct `looks_like` sprites in both states.

#### Additional context
PR labels C++, Documentation and Markdown are spurious, they were automatically added by the bot because at one point this PR had a commit (caf64cd2d91fc64319aafc8338ef1268e66ae3a2) that squash merged #62945 for testing and ensuring that checks would pass.

### CODE REVIEW

File: data/json/items/tool/fishing.json [hunk 1]
```diff
@@ -100,6 +100,93 @@
     "qualities": [ [ "FISHING", 2 ] ],
     "flags": [ "FISH_GOOD", "SHEATH_SPEAR" ]
   },
+  {
+    "id": "fishing_rod_2pc_packed",
+    "type": "TOOL",
+    "name": { "str": "two-piece fishing rod (packed)" },
```
Comment (TheShadowFerret) [inline data/json/items/tool/fishing.json:106]:
```suggestion
    "name": { "str": "two-piece fishing rod (packed)", "str_pl": "two-piece fishing rods (packed)" },
```

File: data/json/items/tool/fishing.json [hunk 2]
```diff
@@ -100,6 +100,93 @@
...
+      "menu_text": "Disassemble",
+      "type": "transform",
+      "target": "fishing_rod_2pc_packed",
+      "msg": "You take apart your two-piece fishing rod."
+    }
+  },
+  {
+    "id": "fishing_rod_tele_packed",
+    "type": "TOOL",
+    "name": { "str": "telescoping fishing rod (packed)" },
```
Comment (TheShadowFerret) [inline data/json/items/tool/fishing.json:150]:
```suggestion
    "name": { "str": "telescoping fishing rod (packed)", "str_pl": "telescoping fishing rods (packed)" },
```

**Review by TheShadowFerret — CHANGES_REQUESTED** [review]

**Comment by cake-pie** [comment]
This PR [passed checks](https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/3982607010) at caf64cd which included squashed merge of #62945 for testing purposes, so I've taken out the extra commit and declaring this ready to go whenever that also gets merged.

**Comment by cake-pie** [comment]
Rebased to resolve conflicts. No longer blocked by #62945; ready to merge.

**Pull request merged**